rust/kernel/file_operations.rs

```diff
@@ -400,6 +405,39 @@ impl IoctlCommand {
     }
 }
 
+/// Trait for extracting file open arguments from kernel data structures.
+///
+/// This is meant to be implemented by registration managers.
+pub trait FileOpenAdapter {
```

Comment:
Does this need to be `pub` then? i.e. all managers are expected to be inside `rust/kernel/`?

Reply:
Yes, it needs to be public. Binder has a file system (in C) that has its own registration scheme that only makes sense in that context, that is, that adapter will be specific to binder so it won't live in `rust/kernel`.

I'll probably need to make `FileOperationsVtable::build` public as well when I hook that up.